langworld_db_data/featureprofiletools/value_renamer.py

```diff
@@ -40,13 +59,57 @@ def rename_value_in_profiles_and_inventories(
                 output_dir=self.output_feature_profiles_dir,
             )
 
-        if not self.output_inventories_dir.exists():
-            self.output_inventories_dir.mkdir(parents=True, exist_ok=True)
-        self._update_features_listed_values(
-            id_of_value_to_rename=id_of_value_to_rename,
-            new_value_name=new_value_name,
-            input_file=self.input_inventories_dir / "features_listed_values.csv",
-            output_file=self.output_inventories_dir / "features_listed_values.csv",
+    def _value_id_exists(
+        self,
+        id_of_value_to_rename: str,
+    ) -> bool:
+        data_from_file = read_dicts_from_csv(
+            self.input_inventories_dir / "features_listed_values.csv"
+        )
+        for line in data_from_file:
+            if line["id"] == id_of_value_to_rename:
+                return True
+        return False
+
+    def _current_value_name_is_equal_to_new_value_name(
+        self,
+        id_of_value_to_rename: str,
+        new_value_name: str,
+    ):
+        data_from_file = read_dicts_from_csv(
+            self.input_inventories_dir / "features_listed_values.csv"
+        )
```

Comment:
note that you have repetition here.

I suggest creating an attribute in `__init__()` that contains all lines from file with listed values.

For example:
```py
self.inventory_of_listed_values = read_dicts_from_csv(
            self.input_inventories_dir / "features_listed_values.csv"
        )
```
Then you just use it in these two private methods

Reply:
Done